app/controllers/nodes_controller.rb

```diff
@@ -16,6 +18,18 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  def bootstrap
+    salt = Pharos::Salt.new
+    available_roles = %i(master)
+    salt.minions.keys.each do |minion|
+      Minion.transaction do
+        Minion.find_or_create_by!(hostname: minion).assign_role(available_roles)
```

Comment:
This will need to change as soon as we merge [this](https://github.com/SUSE/pharos/pull/5). We should only create a Minion in the database when we have information that this minion exists (e.g. when we get a minion-start event). That said, we only need to apply the master role to an existing minion (unless we have none).